### PR TITLE
Add support for handling property bindings in the yaml writer

### PR DESCRIPTION
**Pull request by jrmuizel**
This doesn't try and do anything fancy, it just writes out the current
value. The code is very similar to the code from webrender/scene.rs but
probably not worth sharing, because it's pretty trivial.

---
This change is [Reviewable](https://reviewable.io/reviews/servo/webrender/1996)

### CODE REVIEW

File: wrench/src/scene.rs [hunk 1]
```diff
@@ -5,6 +5,76 @@
...
+/// Stores a map of the animated property bindings for the current display list. These
+/// can be used to animate the transform and/or opacity of a display list without
+/// re-submitting the display list itself.
+pub struct SceneProperties {
+    transform_properties: HashMap<PropertyBindingId, LayoutTransform>,
+    float_properties: HashMap<PropertyBindingId, f32>,
+}
+
+impl SceneProperties {
+    pub fn new() -> SceneProperties {
```
Comment (kvark) [inline wrench/src/scene.rs]: would be easier to just put `#[derive(Default)]` on top of the struct

File: wrench/src/scene.rs [hunk 2]
```diff
@@ -5,6 +5,76 @@
...
+        }
+
+        for property in &properties.floats {
+            self.float_properties
+                .insert(property.key.id, property.value);
+        }
+    }
+
+    /// Get the current value for a transform property.
+    pub fn resolve_layout_transform(
```
Comment (kvark) [inline wrench/src/scene.rs]: so why can't we share this code with the `scene.rs`? It's not complex, but still a few pages

File: wrench/src/scene.rs [hunk 3]
```diff
@@ -16,6 +86,7 @@ pub struct ScenePipeline {
 
 /// A complete representation of the layout bundling visible pipelines together.
 pub struct Scene {
+    pub properties: SceneProperties,
```
Comment (kvark) [inline wrench/src/scene.rs]: this `Scene` struct could also benefit from `Default` being derived

File: wrench/src/yaml_frame_writer.rs [hunk 4]
```diff
@@ -179,14 +179,10 @@ fn maybe_radius_yaml(radius: &BorderRadius) -> Option<Yaml> {
     }
 }
 
-fn write_sc(parent: &mut Table, sc: &StackingContext, filter_iter: AuxIter<FilterOp>) {
+fn write_sc(parent: &mut Table, sc: &StackingContext, scene: &Scene, filter_iter: AuxIter<FilterOp>) {
```
Comment (kvark) [inline wrench/src/yaml_frame_writer.rs]: nit: looks like we actually only need the properties here and below, not the whole scene

File: wrench/src/scene.rs (hunk 2 above)
Comment (jrmuizel) [reply wrench/src/scene.rs]: It's really just setting and getting from hash tables, and there are differences in the the signatures of the resolve methods due to the kind of references passed in. Also one piece of code uses warn! vs println!. Sharing it really just seems like a hassle.

**Comment by glennw** [comment]
I'm not sure if there's further follow ups needed to this, but I figure we can do them after merge if required, so I'll get this merged while the travis CI queue is relatively empty.

@bors-servo r+

**Pull request merged**